cmd/zpool/zpool_main.c

```diff
@@ -9466,21 +9479,30 @@ zpool_do_set(int argc, char **argv)
 
 	return (error);
 }
-/* Add up the total number of bytes left to initialize across all vdevs */
+
+/* Add up the total number of bytes left to initialize/trim across all vdevs */
 static uint64_t
-vdev_initialize_remaining(nvlist_t *nv)
+vdev_activity_remaining(nvlist_t *nv, zpool_wait_activity_t activity)
 {
 	uint64_t bytes_remaining;
 	nvlist_t **child;
 	uint_t c, children;
 	vdev_stat_t *vs;
 
+	ASSERT(activity == ZPOOL_WAIT_INITIALIZE ||
+	    activity == ZPOOL_WAIT_TRIM);
```

Comment:
Please use `assert` instead of `ASSERT` in the user space commands and libraries.

Reply:
@behlendorf Do you know why we should not use `ASSERT`?  Should we make it not compile?

Reply:
> Should we make it not compile?

Yes, we probably should.  From what I've seen the original code was quite good about using `assert(3)` for the userspace libraries and binaries.  As I understand it the ASSERT/VERIFY macros we provided as compatibility wrappers on all platforms solely to allow building the kernel code in user space (`libzpool`, etc).

Using them elsewhere doesn't lead to any problems, but it does muddle the long standing layering and can be confusing when looking at a source file which uses both.  If you look at `libzfs` you can see they've been slowing sneaking in over time.